src/stdlib/sieve.rs

```diff
@@ -0,0 +1,196 @@
+use std::collections::HashSet;
+
+use crate::compiler::prelude::*;
+
+fn sieve(
+    value: Value,
+    permitted_characters: Value,
+    replace_single: Value,
+    replace_repeated: Value,
+) -> Resolved {
+    let value = value.try_bytes_utf8_lossy()?;
+    let replace_single = replace_single.try_bytes_utf8_lossy()?;
+    let replace_repeated = replace_repeated.try_bytes_utf8_lossy()?;
+
+    match permitted_characters {
+        Value::Bytes(bytes) => {
+            let string = String::from_utf8_lossy(&bytes);
+            let characters: HashSet<char> = string.chars().collect();
+            let mut result = String::with_capacity(value.len());
+            let mut missed_length = 0;
+            for char in value.chars() {
+                if characters.contains(&char) {
```

Comment:
Is there any crate that uses a more efficient algorithm for string matching?

Reply:
I will check it out and let you know.

Reply:
Btw, this is an optional step, I was curious. If there's nothing mature available we can just leave a comment that this can be optimized.

Reply:
Sorry, I forgot to respond. I was thinking, maybe it would make sense to remove the option to use `string` and enforce only `regex`. On that one at least no manual checks are performed and I just iterate through the returned matches.

Reply:
Sounds reasonable to me